src/loaders/lottie/tvgLottieBuilder.cpp

```diff
@@ -1071,21 +1071,26 @@ void LottieBuilder::updateText(LottieLayer* layer, float frameNo)
                     else if ((*s)->based == LottieTextRange::Based::Lines) basedIdx = line;
 
                     if (basedIdx < start || basedIdx >= end) continue;
-                    auto& matrix = shape->transform();
 
                     shape->opacity((*s)->style.opacity(frameNo));
 
                     auto color = (*s)->style.fillColor(frameNo);
                     shape->fill(color.rgb[0], color.rgb[1], color.rgb[2], (*s)->style.fillOpacity(frameNo));
 
+                    Matrix matrix;
+                    identity(&matrix);
```

Comment:
The shape's transform will be reset on every iteration. This means that during the text range iteration process (line 1064), any previously applied transform to the shape might be overwritten, results in a meaningless job. It seems the text-range logic is getting messed up. Could you please double-check this?

Reply:
yes, this is intentional. if we don;t overwrite previously applied transformations, this is what is happening:
 
![rotBefore](https://github.com/user-attachments/assets/a3f221fb-ef83-4c8c-9021-229e91584dbf)

this should be a static image, but each time we update text, one extra transformation is added to the previous one. 

I could use shape->rotate/scale/translate so that I don;t need to create matrix object, but than we need to update scale api to expect x and y scaling factors, not only one value. this will overwrite the transform matrix internally.

Reply:
This is what I see the code.
```
shape = gen();
for(; ; ;) {
  matrix m;
  shape->transform(m);   //useless effect during iteration.
}
scene->push(shape);
```

Reply:
@hermet please check now. you were right, that combining multiple transformations while iterating over ranges was not handled properly. 
as I tested in AE, all parameters (like rotation, scaling and translation) are cumulated first, and then applied. 

sample: 
[multiRanges.json](https://github.com/user-attachments/files/17491121/multiRanges.json)

result (rotate anchor point is not perfect now - that's why slight differences are still visible)
<img width="954" alt="Zrzut ekranu 2024-10-23 o 19 13 16" src="https://github.com/user-attachments/assets/ded3e41f-1e4c-44a9-b7e6-f6e862e65256">